### ISSUE TITLE
Copy/Paste Issue

### ISSUE THREAD

**Issue by Komanachi** [brave/brave-browser#41064]
### Description

When I copy some of my passwords via icon on the webui, the characters on the copied password coverts to the lower letters. I mean, I cannot copy and paste original password. If I make the password visible and then copy it manually via Ctrl + C or right click, I can paste it as original state without changing the letters.

The problem occurs on vault.bitwarden.com copy function via icon. I tried different browsers and there is no any problem.

Device: Apple M2 Max
OS: Sonoma 14.6.1
Brave:  2024.9.0

This problem that I face has started 10 or 12 days ago. Before that, there was no any issue. The problem occurs only one password that included ":", "^", and ",". For others, the characters stay same as expected. But for this password, let's say the password is "ASDASD:asd^ASD,ASD". When I manually copy the password there is no problem. But if I use the automatic copy function via icon on the website, it is converted to "asdasd:asd^ASD,ASD".

On the developer tools > console there is no any error. It's hard to explain the issue. I hope it's clear.

### Steps to reproduce

1. go vault.bitwarden.com
2. create a password ASDASD:asd^ASD,ASD
3. then copy it via copying icon
4. paste it somewhere and you will get the result like asdasd:asd^ASD,ASD

### Actual result

asdasd:asd^ASD,ASD

### Expected result

ASDASD:asd^ASD,ASD

### Reproduces how often

Easily reproduced

<div id="outer" style="margin-inline: auto; margin-top: 10px; min-width: 800px; width: 1152px; color: rgb(32, 33, 36); font-family: system-ui, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div id="logo" style="float: right; margin-inline-start: 40px; text-align: end; width: 180px;"><picture><img alt="Brave logo" src="chrome://theme/IDR_PRODUCT_LOGO"></picture><div id="company" style="font-size: 0.7em; text-align: end;">Brave Software Inc</div><div id="copyright" style="font-size: 0.7em; text-align: end;">Copyright © 2024 The Brave Authors. All rights reserved.</div></div>
Brave | 1.69.168 Chromium: 128.0.6613.138 (Official Build) (arm64)
-- | --
Revision | 6918a99aa67f8b2c8cfdc9a44e6ad4f9be6c5be9
OS | macOS Version 14.6.1 (Build 23G93)
JavaScript | V8 12.8.374.31
User Agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36
Command Line | /Applications/Brave Browser.app/Contents/MacOS/Brave Browser --disable-domain-reliability --enable-dom-distiller --enable-distillability-service --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --lso-url=https://no-thanks.invalid --sync-url=https://sync-v2.brave.com/v2 --variations-server-url=https://variations.brave.com/seed --variations-insecure-server-url=https://variations.brave.com/seed --flag-switches-begin --flag-switches-end --component-updater=url-source=https://go-updater.brave.com/extensions
Executable Path | /Applications/Brave Browser.app/Contents/MacOS/Brave Browser
Profile Path | /Users/melihkarasu/Library/Application Support/BraveSoftware/Brave-Browser/Profile 3
Linker | lld
Active Variations | main@91b8a9d0e33034f2368c6ed2015474b1d818a902AllowCertainClientHintsStudy:EnabledBraveAIChatEnabledStudy:EnabledBraveAdblockMobileNotificationsListDefault:EnabledBraveAdsAdEventStudy:EnabledBraveAdsConversionsStudy:EnabledBraveAdsExclusionRulesStudy:EnabledBraveAdsNewTabPageAdsStudy:EnabledBraveAdsSiteVisitStudy:EnabledBraveAdsTextClassificationPageProbabilitiesStudy:EnabledBraveAggressiveModeRetirementExperiment:DisabledBraveAutoTranslateStudy:AutoTranslateDisabledBraveCleanupSessionCookiesOnSessionRestore:DisabledBraveDebounceStudy:EnabledBraveFeedUpdateStudy:EnabledBraveForgetFirstPartyStorage:EnabledBraveGoogleSignInPermissionStudy:EnabledBraveHorizontalTabsUpdateEnabledStudy:EnabledBraveP3AConstellationEnabled:EnabledBraveP3ANebulaRelease:EnabledBraveP3ATypicalJSONDeprecationEnabled:EnabledBraveRewardsAllowSelfCustodyProvidersStudy:EnabledBraveRewardsWebUiPanelStudy:EnabledBraveScreenFingerprintingBlockerStudy:EnabledBraveSearchAdStudy:EnabledBraveWalletAnkrBalancesEnabled:DefaultBraveWebcompatExceptionsServiceReleaseStudy:EnabledClampPlatformVersionClientHint:DisabledCookieListDefaultStudy:EnabledCrossPlatformVPNStudy:EnabledDefault1pBlockingStudy:DisabledDisableReduceLanguage:EnabledEphemeralStorageStudy:EnabledExtensionsManifestV2StudyRelease:EnabledMacCoreLocationBackendStudy:EnabledPartitionedCookies:EnabledSpeedreader TTS:EnabledUndecryptablePasswords:EnabledUserActivityStudy:TriggersWhatsNewStudy:Enabled

</div><br class="Apple-interchange-newline">

a

### Channel information

- [X] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [X] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [X] in the latest version of Chrome

**Comment by fmarier** [comment]
I was able to reproduce in Brave but not in Chrome using this minimal test case: https://fmarier.github.io/brave-testing/clipboard.html

Here are a list of test cases, only the first one is lowercasing the `A`:
- `A:^C,D`: broken
- `A:C,D`: okay
- `A:^,`: okay
- `A:^C,`: okay
- `A:^,D`: okay
- `A:^CD`: okay
- `A^CD`: okay

but only when copying to the clipboard using the JS API. The keyboard shortcut `Ctrl+c` works as expected.

**Comment by fmarier** [comment]
The fact that it works as expected in Chrome suggests to me that we're doing something weird with the clipboard.

The only thing I can think of is https://github.com/brave/brave-browser/issues/33037, but that feature is only supposed to be active on [a small number of sites](https://github.com/brave/adblock-lists/blob/master/brave-lists/clean-urls-permissions.json).

cc @boocmp

**Comment by fmarier** [comment]
Another test case from https://community.brave.com/t/text-changes-when-using-clipboard-writetext/566397

- `https://www.abc.com/<br>abc` becomes `https://www.abc.com/%3Cbr%3Eabc`

**Comment by fmarier** [comment]
It does like like it's related to copy-clean-link because it can also be reproduced right here by:
1. Selecting `https://www.abc.com/<br>abc` in the [comment above](https://github.com/brave/brave-browser/issues/41064#issuecomment-2359498482).
2. Right-clicking on the selection.
3. Choosing "copy clean link"
4. Pasting in a new comment here

**Comment by fmarier** [comment]
Looks like there's two problems here:
1. copy-clean-link is doing something weird when detecting URLs.
2. copy-clean-link is active when it's not supposed to.

**Comment by kjozwiak** [comment]
The above requires `1.70.119` or higher for `1.70.x` verification 👍

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.70.119 Chromium: 129.0.6668.70 (Official Build) (arm64)
-- | --
Revision | a15c836a4df987f118ece1645f54b081019049de
OS | macOS Version 15.0 (Build 24A335)

- Verified steps from issue description and [#41064 (comment)](https://github.com/brave/brave-browser/issues/41064#issuecomment-2359498482)
- Verified creating new entry on vault.bitwarden.com with `ASDASD:asd^ASD,ASD` as the password, copy pasting it from context menu works as expected, Doesn't change the case of letters
- Verified visiting `https://fmarier.github.io/brave-testing/clipboard.html` and ensured that `A:^C,D` was copied/pasted
- Verified visiting `https://www.abc.com/%3Cbr%3Eabc`in a new tab redirects and loads `https://www.abc.com/<br>abc`

https://github.com/user-attachments/assets/ff3b1409-1ede-40fb-89b3-9619169eea68

**Comment by GeetaSarvadnya** [comment]
Verification **PASSED** on

```
Brave | 1.70.119 Chromium: 129.0.6668.70 (Official Build) (64-bit)
-- | --
Revision | a15c836a4df987f118ece1645f54b081019049de
OS | Windows 10 Version 22H2 (Build 19045.4894)
```
- Verified the description from the issue https://github.com/brave/brave-browser/issues/41064#issue-2525717014
- ensured that the correct text/strings/links were being copied as per the following:

* visited vault.bitwarden.com and created a new entry with `ASDASD:asd^ASD,ASD` as the password
* once the new entry was created, ensured that copying via the icon copied/pasted `ASDASD:asd^ASD,ASD`
* visited https://fmarier.github.io/brave-testing/clipboard.html and ensured that `A:^C,D` was copied/pasted

**Comment by btlechowski** [comment]
Verified with

Brave | 1.70.119 Chromium: 129.0.6668.70 (Official Build) (64-bit)
-- | --
Revision | a15c836a4df987f118ece1645f54b081019049de
OS | Linux


Reproduced the original issue using 1.70.114 Chromium: 129.0.6668.59 with the STR/Cases outlined via https://github.com/brave/brave-browser/issues/41064#issue-2525717014 & https://fmarier.github.io/brave-testing/clipboard.html.

Verified fixed on 1.70.119

* visited vault.bitwarden.com and created a new entry with `ASDASD:asd^ASD,ASD` as the password
* once the new entry was created, ensured that copying via the icon copied/pasted `ASDASD:asd^ASD,ASD`
* visited https://fmarier.github.io/brave-testing/clipboard.html and ensured that `A:^C,D` was copied/pasted

![35965_1](https://github.com/user-attachments/assets/a21d365b-65ef-4a0c-b6f7-db624526197a)|![35965_2](https://github.com/user-attachments/assets/110a56dc-f481-4e59-aef8-23ff20c21547)
--|--
